test_suite/generic/helpers.py

```diff
@@ -1,10 +1,10 @@
-from test_suite import conftest
+import os
+import json
 
 
-def instance_data(host):
-    values_to_find = [host.backend.hostname]
-    values_to_find.extend(host.addr(host.backend.hostname).ipv4_addresses)
-
-    return conftest.__get_instance_data_from_json(
-        key_to_find='address', values_to_find=values_to_find
-    )
+def __get_instance_data_from_json(key_to_find, values_to_find):
+    with open(os.environ['CIV_INSTANCES_JSON'], 'r') as f:
```

Comment:
We can make it even more generic and pass the json file path as function arg.

Reply:
I'm not sure I understand why this change is better than the existing one , I will have to pass more arguments to get the same instance data that we got before.
If we need some other function to make some specific operation I would suggest to do that separately and not mix it with this one that should only get the instance data, or maybe I just didn't understand the purpose of this idea?

Reply:
Ok, a few things:
- I didn't say it was better. It was just a suggestion.
- If it's a no-go for me I would have marked the pull requests as "needs work / request changes" again. But I just commented on it. So feel free to reject these comments without any further explanation, if that's the case.
- I think you understood the opposite. I was suggesting the change to make the function actually less specific, instead of having the env var reference inside there.
- This follows the "Separation of concerns" principle which is basically making sure the modules and functions only mind their business. A "helper" module shouldn't have this kind of references to env vars. Again, that's just a point of view, not a mandate.

In summary, if you think my comment does not make sense, you can always sync with me via chat, and feel free to also "resolve" the comments right away, when they don't indicate a showstopper.

Reply:
Oh ok I didn't notice to this line "with open(instances_json_path, 'r') as f:" this is why I didn't understand what's the benefit from this :) cause then it showed this one : "with open(os.environ['CIV_INSTANCES_JSON'], 'r') as f:" as it supposed to stay and not removed.

In general I agree about this, but since it is CIV Project so I don't think other instance json file should be handled here.

But to make it modular as you suggested I would create a configuration veriable in the helpers file since it's permenent variable for this CIV project and it's only 'CIV_INSTANCES_JSON', instead of adding this each time you call this function.

INSTANCES_JSON_PATH = <path/CIV_INSTANCES_JSON>
def __get_instance_data_from_json(path = 'INSTANCES_JSON_PATH', key_to_find, values_to_find):
with open(path, 'r') as f:
instances_json_data = json.load(f)
for instance in instances_json_data.values():
if key_to_find in instance.keys() and instance[key_to_find] in values_to_find:
return instance

Reply:
LGTM !

Reply:
Fix is in this commit: https://github.com/osbuild/cloud-image-val/pull/316/commits/37a7f1e96de7ddaf921c3d710a975463d04626a3

Verified and it passed.
Ack if you approve this and merge:)

Reply:
Approved!